src/Base/Core/Quality/IQuality.hs

```diff
@@ -48,6 +46,14 @@ instance Show Quality where
   show (Diminished i) = if i == 1 then "dim" else show i ++ "dim"
   show (Augmented i)  = if i == 1 then "aug" else show i ++ "aug"
 
+
+iQualFrom :: Quality -> Maybe Quality
+iQualFrom (Diminished x) = if x > 0 && x < 12 then Just (Diminished x)
+                           else Nothing
+iQualFrom (Augmented x) = if x > 0 && x < 12 then Just (Augmented x)
+                           else Nothing
+iQualFrom q = Just q
```

Comment:
So I'm not sure why this function is part of the API. Such a thing shouldn't be necessary with a collection of appropriate smart constructors. I'll probably push some edits to this and the code that uses it rather than make suggestions here, since I'm guessing the changes will permeate fairly deep into the code.

Reply:
Deprecated this function, so resolving these comments.